orchestrator/fork.go

```diff
@@ -0,0 +1,105 @@
+package orchestrator
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	registry "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum-optimism/supersim/config"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+const blockTime = 2
+
+func ConfigFromForkCLIConfig(cfg *config.ForkCLIConfig) (*OrchestratorConfig, error) {
+	superchain := registry.Superchains[cfg.Network]
+	chains := []ChainConfig{}
+
+	// L1
+	l1Client, err := ethclient.Dial(superchain.Config.L1.PublicRPC)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial l1 public rpc: %w", err)
+	}
+
+	l1ForkHeight := big.NewInt(rpc.FinalizedBlockNumber.Int64())
+	if cfg.L1ForkHeight > 0 {
+		l1ForkHeight = new(big.Int).SetUint64(cfg.L1ForkHeight)
+	}
+	l1Header, err := l1Client.HeaderByNumber(context.Background(), l1ForkHeight)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve L1 header: %w", err)
+	}
+
+	chains = append(chains, ChainConfig{
+		Port:           0,
+		ChainID:        superchain.Config.L1.ChainID,
+		Accounts:       DefaultAccounts,
+		Mnemonic:       DefaultMnemonic,
+		DerivationPath: DefaultDerivationPath.String(),
+		ForkConfig: &ForkConfig{
+			RPCUrl:      superchain.Config.L1.PublicRPC,
+			BlockNumber: l1Header.Number.Uint64(),
+		},
+	})
+
+	// L2s
+	l2ChainIds := cfg.Chains
```

Comment:
is it possible the l1 chain id could end up in `cfg.Chains` or would that be caught by the `isChainInSuperchain` check?

Reply:
yep exactly it would be caught in the CLI validation step since it'll be missing the superchain list of chains